contrib/epee/include/net/connection_basic.hpp

```diff
@@ -86,13 +97,15 @@ class connection_basic { // not-templated base class for rapid developmet of som
     /// Socket for the connection.
     boost::asio::ip::tcp::socket socket_;
 
-		std::atomic<long> &m_ref_sock_count; // reference to external counter of existing sockets that we will ++/--
 	public:
 		// first counter is the ++/-- count of current sockets, the other socket_number is only-increasing ++ number generator
-		connection_basic(boost::asio::io_service& io_service, std::atomic<long> &ref_sock_count, std::atomic<long> &sock_number);
+		connection_basic(boost::asio::ip::tcp::socket&& socket, boost::shared_ptr<socket_stats> stats);
 
 		virtual ~connection_basic() noexcept(false);
 
+                //! \return `socket_stats` object passed in construction (ptr never changes).
+		socket_stats& get_stats() const noexcept { return *m_stats; /* verified in constructor */ }
```

Comment:
Probably needs to be const socket_stats& if the function's const.

Reply:
Weirdly enough, it doesn't have to be per C++ rules, but arguably should be. I will make it so.

Reply:
This can't return a `const` instance because it uses a non-const function on the handler. Instead, I made `get_stats` not-const. It can be upgraded/updated in the future.